From: Ping Cheng <pinglinux@gmail.com>
Date: Sun, 24 Jun 2012 23:29:29 -0500
Subject: wacom: do not request tablet data on MT Tablet PC pen interface
Bug-Debian: http://bugs.debian.org/677164

When in commit 1963518b9b1b (Input: wacom - add 0xE5 (MT device)
support, 2012-04-29) the driver stopped asking for multitouch tablet
data on the pen interface of a tablet PC, as a side effect we started
executing the "else" to that if statement.  Oops.

This is needed for the 0xE6 tablet in the Thinkpad x220t to be usable
again.  Meanwhile the 0xE3 works fine without this.  Not sure why. -jn

[jn: extracted from a larger commit in the input-wacom repository,
 with new description]

Signed-off-by: Ping Cheng <pingc@wacom.com>
Signed-off-by: Jonathan Nieder <jrnieder@gmail.com>
---
 drivers/input/tablet/wacom_sys.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/input/tablet/wacom_sys.c b/drivers/input/tablet/wacom_sys.c
index 8b31473a81fe..19e4725858dd 100644
--- a/drivers/input/tablet/wacom_sys.c
+++ b/drivers/input/tablet/wacom_sys.c
@@ -467,6 +467,7 @@ static int wacom_query_tablet_data(struct usb_interface *intf, struct wacom_feat
 		}
 	} else if (features->type <= BAMBOO_PT &&
 		   features->type != WIRELESS &&
+		   features->type != TABLETPC2FG &&
 		   features->device_type == BTN_TOOL_PEN) {
 		do {
 			rep_data[0] = 2;
-- 
1.7.11.rc3

